libp2p/relay/circuit_v2/transport.py

```diff
@@ -154,6 +168,36 @@ async def dial_peer_info(
             If the connection cannot be established
 
         """
+        # Prefer stored /p2p-circuit addrs from peerstore
+        # Try first to read addresses from peerstore
+        peer_store = self.host.get_peerstore()
+        stored_addrs = peer_store.addrs(peer_info.peer_id)
```

Comment:
`peer_store.addrs` will raise a `PeerStoreError` if the peer we're dialing isn't already in the peerstore, making dialing new peers impossible. Maybe wrap in a `try/except` or similar?